test/test.js

```diff
@@ -0,0 +1,78 @@
+const autoprefixer = require("../");
+const chai = require("chai");
+const parse = require("css-parse");
+const fs = require("fs");
+const path = require("path");
+const stylus = require("stylus");
+
+const test_path = path.join(__dirname, "fixtures");
+
+const should = chai.should();
+
+const match_expected = (file, args, done) => {
+  stylus(fs.readFileSync(path.join(test_path, file), "utf8"))
+    .use(autoprefixer(args))
+    .render((err, css) => {
+      if (err) {
+        return done(err);
+      }
+
+      const expected = fs.readFileSync(
+        path.join(test_path, file.replace(".styl", ".css")),
+        "utf8"
+      );
+
+      parse(css).should.eql(parse(expected));
+
+      done();
+    });
+};
+
+describe("basic", function() {
+  it("works", function(done) {
+    match_expected("basic.styl", null, done);
+  });
+
+  it("doesn't bail when given whack arguments", function(done) {
+    match_expected(
+      "basic.styl",
+      {
+        foo: "bar"
+      },
+      done
+    );
+  });
+
+  it("takes browser options", function(done) {
+    match_expected(
+      "browser.styl",
+      {
+        browsers: ["last 2 versions"],
+        grid: true
+      },
+      done
+    );
+  });
+
+  it("returns correct sourcemaps", function(done) {
+    const filename = path.join(test_path, "basic.styl");
+
+    const style = stylus(fs.readFileSync(filename, "utf8"))
+      .set("sourcemap", true)
+      .set("filename", filename)
+      .use(autoprefixer());
+
+    style.render((err, css) => {
+      if (err) {
+        return done(err);
+      }
+
+      style.sourcemap.should.be.an("object");
+      style.sourcemap.sources[0].should.equal("stylus");
+      style.sourcemap.version.should.equal(3);
+      style.sourcemap.mappings.should.equal("AAAA;EACE,cAAA");
+
+      done();
+    });
+  });
+});
```

Comment:
Would prefer to commit a `package-lock.json` rather than a yarn lock, as npm is the more frequently used option and the default for node.

Reply:
Good catch, it shouldn't actually even be here! Apps should manage their lock file and not libraries, I'll remove it entirely.